File: frontend/src/services/chatService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api';

const chatService = {
    sendMessage: async (message) => {
        const response = await axios.post(`${API_URL}/chat`, { message });
        return response.data.response;
    }
};

export default chatService;

File: mobile/lib/data/services/chat_service.dart
import 'package:dio/dio.dart';
import '../../core/constants/api_constants.dart';
import 'api_service.dart';

class ChatService {
  final ApiService _apiService = ApiService();

  Future<String> sendMessage(String message) async {
    try {
      final response = await _apiService.dio.post(
        ApiConstants.chatEndpoint,
        data: {'message': message},
      );

      if (response.statusCode == 200 && response.data != null) {
        return response.data['response'] ?? 'No response';
      }
      return 'Error: Invalid response from server';
    } catch (e) {
      print('Chat Error: $e');
      return 'Désolé, je ne peux pas répondre pour le moment. ($e)';
    }
  }
}
